javascript/javascript/JavaScriptLexer.g4

```diff
@@ -170,7 +170,8 @@ Await:                          'await';
 /// when parsing strict mode
 
 Implements:                     'implements' {this.IsStrictMode()}?;
-Let:                            'let' {this.IsStrictMode()}?;
+StrictLet:                      'let' {this.IsStrictMode()}?;
+NonStrictLet:                   'let' {!this.IsStrictMode()}?;
```

Comment:
Hi, maybe it's better to use `Let` rule which will be like `Let : StrictLet | NonStrictLet`, WDYT?

Reply:
Oh right, makes sense!

Reply:
Nice, LGTM

Reply:
#1591 has simillar problem

Reply:
@loonydev I don't think it's possible. Firstly, it can be simplified by jus `Let: "let'`. Secondly, unified `Let` token substitutes `StrictLet` and `NonStrictLet`, and it becomes impossible to distinguish between `StringLet` and `NonStrictLet` in parser (`NonStrictLet` used in `identifier` rule).

Reply:
In case where we have both of them, it's better to have one Let. But you right about simplification.